From 3a947fa318e50db950609d319523e495995b98f6 Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Wed, 23 Nov 2016 14:48:17 -0500
Subject: [PATCH] menu: Don't leak check menu items

Pointed out in
https://bugzilla.gnome.org/show_bug.cgi?id=774686

Funtoo maintainer note: patch has been rebased to correctly patch 
gnome-3.20.9-r1.ebuild. The contents of the patch are equivalent
to the git commit.

---
--- a/gtk/gtkmenu.c	2017-07-19 02:53:36.324025475 +0000
+++ b/gtk/gtkmenu.c	2017-07-19 02:54:36.028025813 +0000
@@ -3136,6 +3136,8 @@
       max_toggle_size = indicator_width;
 
       gtk_widget_destroy (menu_item);
+      gtk_widget_ref_sink (menu_item);
+      gtk_object_unref (menu_item);
     }
 
   min_width += 2 * max_toggle_size + max_accel_width;
